File: dev/20190902/k2works/app/FizzBuzz/Application/Command/FizzBuzzValueCommand.cs
using FizzBuzz.Application.Service;

namespace FizzBuzz.Application.Command
{
    public class FizzBuzzValueCommand: IFizzBuzzCommand
    {
        private IFizzBuzzType _type;

        public FizzBuzzValueCommand(IFizzBuzzType type)
        {
            _type = type;
        }
        public IFizzBuzz Execute(int number)
        {
            return _type.Generate(number);
        }
    }
}

File: dev/20190531/session/api/fizzbuzz/src/main/java/fizzbuzz/application/command/FizzBuzzValueCommand.java
package fizzbuzz.application.command;

import fizzbuzz.application.service.IFizzBuzz;
import fizzbuzz.application.service.IFizzBuzzCommand;
import fizzbuzz.domain.type.FizzBuzzType;

public class FizzBuzzValueCommand implements IFizzBuzzCommand {
  private FizzBuzzType _type;

  public FizzBuzzValueCommand(FizzBuzzType type) {
    _type = type;
  }

  @Override
  public IFizzBuzz execute(int number) {
    return _type.generate(number);
  }

}

